ethereum/stratum/src/test/java/org/hyperledger/besu/ethereum/stratum/StratumServerTest.java

```diff
@@ -47,16 +72,194 @@ public void runJobSetZeroDifficultyShouldNotThrow() {
+    stratum
+        .start()
+        .onComplete(
+            testContext.succeeding(
+                server -> {
+                  WebClient http =
+                      WebClient.create(
+                          vertx,
+                          new WebClientOptions()
+                              .setDefaultHost(host)
+                              .setDefaultPort(server.actualPort()));
+                  http.put("/")
+                      .sendBuffer(DEFAULT_TEST_MESSAGE)
+                      .onComplete(
+                          testContext.succeeding(
+                              resp -> {
+                                testContext.verify(
+                                    () -> {
+                                      assertThat(resp.statusCode()).isEqualTo(405);
+                                      stratum
+                                          .stop()
+                                          .onComplete(testContext.succeedingThenComplete());
+                                    });
+                              }));
+                }));
+  }
+
+  @Test
+  public void handlePlainText(final Vertx vertx, final VertxTestContext testContext) {
+    String host = "localhost";
+    StratumServer stratum = new StratumServer(vertx, mockPoW, 0, host, "", new NoOpMetricsSystem());
+    stratum
+        .start()
+        .onComplete(
+            testContext.succeeding(
+                server -> {
+                  NetClient client = vertx.createNetClient();
+                  client
+                      .connect(server.actualPort(), host)
+                      .onComplete(
+                          testContext.succeeding(
+                              socket ->
+                                  socket
+                                      .handler(
+                                          resp ->
+                                              testContext.verify(
+                                                  () -> {
+                                                    assertThat(resp.length()).isNotZero();
+                                                    stratum
+                                                        .stop()
+                                                        .onComplete(
+                                                            testContext.succeedingThenComplete());
+                                                  }))
+                                      .write(
+                                          DEFAULT_TEST_MESSAGE.appendString(System.lineSeparator()))
+                                      .onComplete(
+                                          writeResult ->
+                                              testContext.verify(
+                                                  () ->
+                                                      assertThat(writeResult.failed())
```

Comment:
please check what is being returned? Same as other comments, check the body matches expectations

Reply:
I added the assertion on the body here too